pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/AbstractRule.java

```diff
@@ -503,4 +503,48 @@ public int hashCode() {
         return getClass().getName().hashCode() + (getName() != null ? getName().hashCode() : 0)
                 + getPriority().hashCode() + (propertyValues != null ? propertyValues.hashCode() : 0);
     }
+    
+    @SuppressWarnings("unchecked")
+    @Override
+    public Rule deepCopy() {
+        Rule rule = null;
+        try {
+            rule = getClass().newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            // Can't happen... we already have an instance
+        }
+        rule.setName(getName());
+        rule.setLanguage(getLanguage());
+        rule.setMinimumLanguageVersion(getMinimumLanguageVersion());
+        rule.setMaximumLanguageVersion(getMaximumLanguageVersion());
+        rule.setSince(getSince());
+        rule.setMessage(getMessage());
+        rule.setRuleSetName(getRuleSetName());
+        rule.setExternalInfoUrl(getExternalInfoUrl());
+        if (usesDFA()) {
+            rule.setUsesDFA();
+        }
+        if (usesTypeResolution()) {
+            rule.setUsesTypeResolution();
+        }
+        if (usesMetrics()) {
+            rule.setUsesMetrics();
+        }
+        rule.setDescription(getDescription());
+        for (final String example : getExamples()) {
+            rule.addExample(example);
+        }
+        rule.setPriority(getPriority());
+        for (final PropertyDescriptor<?> prop : getPropertyDescriptors()) {
```

Comment:
I'm now using `hasDescriptor`, but do not use `copyPropertyDescriptors` (it just creates a new list with the same property descriptor instances), but I'd rather call the `definePropertyDescriptor` method to allow rules overriding the method applying their logic rather than just assigning the list.